webpack/components/TypeAhead/pf4Search/TypeAheadInput.js

```diff
@@ -0,0 +1,30 @@
+import React, { useRef } from 'react';
+import { TextInput } from '@patternfly/react-core';
+
+import useEventListener from '../../../utils/useEventListener';
+import { commonInputPropTypes } from '../helpers/commonPropTypes';
+
+const TypeAheadInput = ({ onKeyPress, onInputFocus, passedProps }) => {
+  const inputRef = useRef(null);
+  const { onChange, ...downshiftProps } = passedProps;
+
+  // What patternfly4 expects for args and what downshift creates as a function is different,
+  // downshift only expects the event handler
+  const onChangeWrapper = (_userValue, event) => onChange(event);
+
+  useEventListener('keydown', onKeyPress, inputRef.current);
+
+  return (
+    <TextInput
+      {...downshiftProps}
+      ref={inputRef}
```

Comment:
I need to dig into why Downshift is requiring a ref.. do you know if there's any other way?  It seems like a React third-party library should work with controlled components and not require you to deal with event listeners and the DOM.

Reply:
For the patternfly 4 version, `TextInput` doesn't have a `onKeyPress` or `onKeyDown` prop ([see here])[https://patternfly-react.surge.sh/documentation/react/components/textinput].

So this means we have to attach a custom event listener and pass it the `onKeyPress` function created by downshift.

I imagine it is the same for the pf3 version using `FormControl`.